concordium-sdk/src/main/java/com/concordium/sdk/exceptions/ED25519Exception.java

```diff
@@ -3,7 +3,7 @@
 import com.concordium.sdk.crypto.ed25519.ED25519ResultCode;
 import lombok.Getter;
 
-public final class ED25519Exception extends Exception {
+public final class ED25519Exception extends RuntimeException {
```

Comment:
What is the reasoning of making this an unchecked exception instead of the prior checked one?

The thing is, when you change this to be an unchecked one then you must document the functions that throws it with java doc. But when it is checked it is also clear from the type what the contract of the function is. 

I am as such not opposed to using an unchecked exception here as there is little the user can do. But please document all functions now throwing this with proper java doc.

Reply:
done